### PR TITLE
Allow all charts to be generated as a dict

### LINKED ISSUE #2360: [FEAT] Allow generation of all visualizations as a dict
### Is your proposal related to a problem?
We are generating some of these charts in a realtime backend and using the `as_dict` parameter on `splink.internals.charts.waterfall_chart` is 3X faster than calling `to_dict` on the altair chart.  For us it is something like 170ms with as_dict vs 670ms with altair.to_dict

### Describe the solution you'd like
We would like the ability to generate all charts as dicts like the `tf_adjustment_chart` allows. 

### Describe alternatives you've considered
we have also used some of the functions in `splink.internals.charts` to take advantage of this prameter now but would prefer to use a supported api.

### PR DESCRIPTION
**Pull request by RossHammer**
### Type of PR

- [ ] BUG
- [X] FEAT
- [ ] MAINT
- [ ] DOC

### Is your Pull Request linked to an existing Issue or Pull Request?

closes #2360

### Give a brief description for the solution you have provided

I have added the `as_dict` parameter to all calls in visualizations that generate a chart.

### PR Checklist

- [X] Added documentation for changes
- [X] Added feature to example notebooks or tutorial (if appropriate)
- [X] Added tests (if appropriate)
- [X] Updated CHANGELOG.md (if appropriate)
- [X] Made changes based off the latest version of Splink
- [X] Run the [linter](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/lint_and_format.html)
- [X] Run the [spellchecker](https://moj-analytical-services.github.io/splink/dev_guides/changing_splink/contributing_to_docs.html?h=spellch#spellchecking-docs) (if appropriate)

### CODE REVIEW

File: splink/internals/linker_components/visualisations.py [hunk 1]
```diff
@@ -75,9 +75,12 @@ class LinkerVisualisations:
     def __init__(self, linker: Linker):
         self._linker = linker
 
-    def match_weights_chart(self) -> ChartReturnType:
+    def match_weights_chart(self, as_dict: bool = True) -> ChartReturnType:
```
Comment (RobinL) [inline splink/internals/linker_components/visualisations.py:78]: `as_dict` should be `False` by default.

i.e. all of the functions in `linker.visualiastions` should return an Altair chart by default, but if the user passes `as_dict=True` they can get back the dict representation without going through altair 

I think this is also the reasons for the test failures

**Review by RobinL — COMMENTED** [review]
Thanks for this - agree it makes sense to for the visualisations to have an `as_dict` arg where possible.  But I think the should always be `False` by default (see comment).

File: splink/internals/linker_components/visualisations.py (hunk 1 above)
Comment (RossHammer) [reply splink/internals/linker_components/visualisations.py]: That is what I intended but must have made a typo, it is updated now when you want to look again

**Review by RobinL — APPROVED** [review]
Thank you!

**Pull request merged**